lib/routes/close_channels/coop_close_channels_dialog.dart

```diff
@@ -0,0 +1,235 @@
+                      style: const TextStyle(color: Colors.blue),
+                      recognizer: TapGestureRecognizer()
+                        ..onTap = () async {
+                          // If device can't handle mailto: link, share the e-mail
+                          try {
+                            await launchLinkOnExternalBrowser(
+                              "mailto:contact@breez.technology",
+                            );
+                          } catch (e) {
+                            final RenderBox box = context.findRenderObject();
+                            final offset =
+                                box.localToGlobal(Offset.zero) & box.size;
+                            final rect = Rect.fromPoints(
+                              offset.topLeft,
+                              offset.bottomRight,
+                            );
+                            Share.share(
+                              "contact@breez.technology",
+                              subject: texts
+                                  .close_channels_dialog_failure_message_middle,
+                              sharePositionOrigin: rect,
+                            );
+                          }
+                        },
+                    ),
+                    TextSpan(
+                      text: texts.close_channels_dialog_failure_message_end,
+                    ),
+                  ],
+                ),
+              ),
+            ],
+          ],
+        ),
+        actions: (dialogMessage.isNotEmpty || hasError == true)
+            ? [
+                TextButton(
+                  style: ButtonStyle(
+                    overlayColor: MaterialStateProperty.resolveWith<Color>((states) {
+                      if (states.contains(MaterialState.pressed)) {
+                        return Colors.transparent;
+                      }
+                      return null; // Defer to the widget's default.
+                    }),
+                  ),
+                  child: Text(
+                    texts.close_channels_dialog_action_close,
+                    style: themeData.primaryTextTheme.labelLarge,
+                  ),
+                  onPressed: () =>
+                      Navigator.of(context).removeRoute(_currentRoute),
+                ),
+              ]
+            : null,
+      ),
+    );
+  }
+
+  Future<void> _closeChannels() async {
+    final texts = context.texts();
+    final themeData = Theme.of(context);
+    try {
+      final accountBloc = AppBlocsProvider.of<AccountBloc>(context);
+      final closeChannelAction = CloseChannelsAction(widget.closingAddress);
+      accountBloc.userActionsSink.add(closeChannelAction);
+      await closeChannelAction.future.then((result) {
+        final closedChannels = result as List<CloseChannelResult>;
+        if (closedChannels.any((c) => !c.isSkipped || !c.failErr.isNotEmpty)) {
```

Comment:
```suggestion
        if (closedChannels.all((c) => !c.isSkipped || !c.failErr.isNotEmpty)) {
```

Reply:
Whoops, had `closedChannels.any((c) => c.isSkipped || c.failErr.isNotEmpty)` in mind when writing it :)
Great catch!

Reply:
Done dc9075fc53f5e4af3dc2b751ff250ee51a9de89f

Reply:
Looks good

Reply:
@JssDWt Actually, I think it was working as expected and in a more efficient way.

This is the case where there aren't any skipped channels or channels with failure errors. Since we're interested in a full success case here, there's no need to check every channel, it's enough to find one bad apple.

Reply:
Reverted to syntax in 

> Whoops, had !closedChannels.any((c) => c.isSkipped || c.failErr.isNotEmpty) in mind when writing it :)
> Great catch!

and added descriptive comment 7d4a6b8f20e1f275e36a77e25cae0bfe40151ef1

Reply:
Hmm, 
`if (closedChannels.any((c) => !c.isSkipped || !c.failErr.isNotEmpty))`
This falls into the if statement if one or more channels were successful. I think you want to fall into this if `all` channels were successful?
Maybe turn it around by doing what you suggested?
`if (!closedChannels.any((c) => c.isSkipped || c.failErr.isNotEmpty))`

Reply:
Done 76244e3ee92d70ce04622088391c2635990bf66a